webportal/src/app/create_item/components/create_data/basic_information.jsx

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  DefaultButton,
+  PrimaryButton,
+  Stack,
+  Text,
+  TextField,
+  FontSizes,
+  FontWeights,
+} from 'office-ui-fabric-react';
+import styled from 'styled-components';
+import { isEmpty } from 'lodash';
+import { TYPE_ENUM } from 'App/utils/constants';
+import { useBoolean } from '@uifabric/react-hooks';
+
+const BasicInformationArea = styled.div`
+  margin-bottom: 50px;
+  margin-left: 100px;
+  margin-right: 100px;
+`;
+
+const BasicInformation = props => {
+  const { state, setState, itemDescription } = props;
+  const [errorMessage, setErrorMessage] = useState(false);
+  const [, updateState] = useState();
+  const forceUpdate = () => updateState({});
+  const [
+    advancedDescription,
+    { toggle: toggleAdvancedDescription },
+  ] = useBoolean(false);
+
+  const columnProps = {
+    tokens: { childrenGap: 'm' },
+    styles: {
+      root: {
+        minWidth: 300,
+        fontSize: FontSizes.mediumPlus,
+        fontWeight: FontWeights.semibold,
+      },
+    },
+  };
+
+  const textStyles = {
+    fieldGroup: { borderRadius: '5px' },
+    subComponentStyles: {
+      label: {
+        root: {
+          fontSize: FontSizes.mediumPlus,
+          fontWeight: FontWeights.semibold,
+        },
+      },
+    },
+  };
+
+  function generateDescription(state, itemDescription) {
+    return (
+      `# ${state.itemObject.name}\n\n${itemDescription.description}\n\n` +
+      (isEmpty(itemDescription.getRawData.value) &&
+      isEmpty(itemDescription.useViaOpenPaiJobSubmition.value) &&
+      isEmpty(itemDescription.useInTheCode.value)
+        ? ''
+        : '## About teh dataset\n\n') +
+      (isEmpty(itemDescription.getRawData.value)
+        ? ''
+        : `### Get raw data\n\n${itemDescription.getRawData.value}\n\n`) +
+      (isEmpty(itemDescription.useViaOpenPaiJobSubmition.value)
+        ? ''
+        : `### Use via OpenPAI job submission\n\n${itemDescription.useViaOpenPaiJobSubmition.value}\n\n`) +
+      (isEmpty(itemDescription.useInTheCode.value)
+        ? ''
+        : `### Use in the code\n\n${itemDescription.useInTheCode.value}\n\n`) +
+      (isEmpty(itemDescription.relatedProject.value)
+        ? ''
+        : `## Related project\n\n${itemDescription.relatedProject.value}\n\n`)
+    );
+  }
+
+  function AdvancedDescriptionTextField({ field, rows }) {
+    return (
+      <TextField
+        label={field.label}
+        multiline={true}
+        rows={rows}
+        value={field.value}
+        onChange={(event, newValue) => {
+          field.value = newValue;
```

Comment:
use setState instead of directly change props

Reply:
Done.